eng/pipelines/typespec-ci.yml

```diff
@@ -1,3 +1,19 @@
+trigger:
+  branches:
+    include:
+    - main
```

Comment:
So you want this run CI on every commit into main? I think that is what this is saying.

Reply:
The pipeline only takes 1-2 minutes so it seemed reasonable to run on every commit to main.  But I also don't mind only running this on a nightly schedule, and either no CI trigger, or scoping the CI trigger to the same paths as the PR trigger.

Reply:
@weshaggard: Would you prefer this pipeline not run on every commit to main, or were you just asking if it was intentional?

Reply:
I'm OK with having it run on every commit for now and if there are issues we can tune it.